src/odemis/gui/cont/tabs.py

```diff
@@ -3409,6 +3416,10 @@ def __init__(self, name, button, panel, main_frame, main_data):
         if not main_data.ebeamControlsMag:
             main_data.ebeam.magnification.subscribe(self._onSEMMag)
 
+    @wxlimit_invocation(1)
+    def _on_tc_stream(self, df, data):
```

Comment:
Right now, there are variables called "fb*", some are called "det*", some are "photods", and here the function mentions "tc". I'd rather try to align the names as much as possible. To keep it generic, I'd suggest something like "fbdet". So here maybe `_on_fbdet2_data()`.